src/client/datascience/kernel-launcher/kernelProcess.ts

```diff
@@ -85,6 +85,33 @@ export class KernelProcess implements IKernelProcess {
 
         let stdout = '';
         let stderr = '';
+        let stdoutProc = '';
+        let stderrProc = '';
+        let exitEventFired = false;
+        const cancelWaiting = new CancellationTokenSource();
+
+        exeObs.proc!.on('exit', (exitCode) => {
+            traceInfo('KernelProcess Exit', `Exit - ${exitCode}`, stderrProc);
+            if (this.disposed) {
+                return;
+            }
+            if (!exitEventFired) {
+                this.exitEvent.fire({ exitCode: exitCode || undefined, reason: stderrProc });
+                exitEventFired = true;
+            }
+            cancelWaiting.cancel();
+        });
+
+        exeObs.proc!.stdout.on('data', (data: Buffer | string) => {
+            stdoutProc += data.toString();
+            traceInfo(`Kernel Output: ${stdoutProc}`);
```

Comment:
Looks like on the stdout data here it's doing an append then a print. Unless I'm reading this wrong. So for each trace error are you going to print the current state as well as all previous stdout?

Reply:
Right, those should just be printing. We only need to append the errors to show them later